backend/src/v2/compiler/argocompiler/argo.go

```diff
@@ -119,6 +130,28 @@ func Compile(jobArg *pipelinespec.PipelineJob, kubernetesSpecArg *pipelinespec.S
 			Entrypoint:         tmplEntrypoint,
 		},
 	}
+	
+	if semaphoreKey != "" {
```

Comment:
no need for this intermediate variable and the block above where they're set. Just use opts.SemaphoreKey

```suggestion
	if opts != nil && opts.SemaphoreKey != "" {
```

Reply:
Done, updated to remove the variable initialization block and used `opts.SemaphoreKey` and `opts.MutexName` directly instead.